### DISCUSSION TITLE
Event Conditions/Actions for Tweens on Cameras

### DISCUSSION THREAD

**Discussion by Silver-Streak** [4ian/GDevelop#2718] [Enhancement ideas (and their technical discussions)]
There had been some discussion on the discord around Tweens and Cameras.

The thought was that since Cameras are accessible (to an extent) as objects, event actions and conditions could be built to utilize Tweens on Camera.

What level of difficulty would there be to get the following, and is this something that could be implemented?

Actions:
- Tween X position of Camera (on a specific layer)
- Tween Y position of Camera (on a specific layer)
- Tween Angle of Camera
- Tween Zoom of Camera

Conditions:
- Tween of Camera (Layer/number/etc) exists
- Tween of Camera is playing
- Tween of Camera is finished

**Comment by arthuro555** [comment]
Generally, I would also add some more non-behavior bound methods, to tween for example scene variables (with those, even if we don't add camera tweens, it should only take one more action to set the camera position to the variable value), or expose the [interpolate](https://jeremyckahn.github.io/shifty/doc/shifty.html#.interpolate) function of shifty (which allows to use tweens via an expression kind of like like lerp).

**Reply by 4ian** [reply]
> tween for example scene variables

We could do this by generalizing the "variables" parameters to be any kind of variables (https://trello.com/c/FwShg063/487-add-support-for-a-generic-variable-parameter-that-would-allow-to-choose-any-variable-object-variable-scene-global-extension-vari).
Though this means here making the extension not only a behavior.

> expose the interpolate function of shifty

Interesting indeed!

**Comment by 4ian** [comment]
> What level of difficulty would there be to get the following, and is this something that could be implemented?

Technically yes but by repeating the existing actions/conditions for cameras... not entirely sure if we want to go this road or rather encourage the pattern of using an object as the invisible camera center?

**Reply by Silver-Streak** [reply]
Using an object is how a lot of people do this now, but that becomes much harder to do if you want to tween say, the angle or zoom.

It also helps avoid issues where a user forgot to add the tween behavior to the object, or used the right click/new editor to select the object for the events (since behavior based events are slightly broken with right-click editing currently, although I think a fix is coming for that?)
